core/src/main/java/nl/nn/adapterframework/pipes/IteratingPipe.java

```diff
@@ -452,18 +475,29 @@ protected MessageOutputStream provideOutputStream(PipeLineSession session) throw
 
 	@Override
 	protected boolean canStreamToNextPipe() {
+		if(getForwards()!=null && (getForwards().get(StopReason.MAX_ITEMS_REACHED.getForwardName())!=null || getForwards().get(StopReason.STOP_CONDITION_MET.getForwardName())!=null)) { // streaming is not possible since the forward is not known before hand
+			return false;
+		}
```

Comment:
This is calculated each time the pipe executes. It is better to calculate it once in configure() and store it in a boolean variable